src/renderer/dx/DxFontRenderData.h

```diff
@@ -70,19 +84,38 @@ namespace Microsoft::Console::Render
                                                                                      DWRITE_FONT_STYLE style,
                                                                                      DWRITE_FONT_STRETCH stretch);
 
-        [[nodiscard]] HRESULT UpdateFont(const FontInfoDesired& desired, FontInfo& fiFontInfo, const int dpi) noexcept;
+        [[nodiscard]] HRESULT UpdateFont(const FontInfoDesired& desired, FontInfo& fiFontInfo, const int dpi, const std::unordered_map<std::wstring_view, uint32_t>& features = {}, const std::unordered_map<std::wstring_view, float>& axes = {}) noexcept;
 
         [[nodiscard]] static HRESULT STDMETHODCALLTYPE s_CalculateBoxEffect(IDWriteTextFormat* format, size_t widthPixels, IDWriteFontFace1* face, float fontScale, IBoxDrawingEffect** effect) noexcept;
 
+        bool DidUserSetFeatures() const noexcept;
+
+        float FontStretchToWidthAxisValue(DWRITE_FONT_STRETCH fontStretch) noexcept;

```

Comment:
I'd recommend making these methods private. We want as less public interfaces as possible, if you know what I mean.

Reply:
Why do we want less public interfaces? What is the benefit?

Reply:
Well, first of all, those methods are not used outside this class. Second I think less public interfaces will help people understand what this class should focus on. My initial idea was that UpdateFont would be the only setter interface in this class, to make it laser-focused on this single job it should be doing.

Reply:
Made them private!